internal/pkg/api/handleCheckin.go

```diff
@@ -337,15 +337,20 @@ func (ct *CheckinT) ProcessRequest(zlog zerolog.Logger, w http.ResponseWriter, r
 	actions, ackToken = convertActions(zlog, agent.Id, pendingActions)
 
 	span, ctx := apm.StartSpan(r.Context(), "longPoll", "process")
+	// ctx, cancel := context.WithTimeout(ctx, pollDuration)
+	// defer cancel()
+
 	if len(actions) == 0 {
 	LOOP:
 		for {
 			select {
 			case <-ctx.Done():
```

Comment:
Currently looking at this the only time this case would be hit would be if the client closes there connection to Fleet Server. As the `span, ctx := apm.StartSpan(r.Context(), ...` is being used as the context here, so then this section writes the response. Looking at this code, it shouldn't even write the response. If the context is cancelled then that means the client is no longer connected.

Reply:
the logic of returning CheckinResponse here was added in this pr: https://github.com/elastic/fleet-server/pull/3165/files#diff-e0c02bac8d151e9941eedd5ef643441665ee3d2f78baf42c121edd45dee08ded

Can the context be cancelled only by the client here? I'm wondering if the `writeResponse` is successful, is it correct to return the `AckToken` without actions? I'm trying to find where the `AckToken` is persisted back to ES.